src/lib/inputs/InputNumber.svelte

```diff
@@ -0,0 +1,397 @@
+
+    const startIncrementing = () => {
+        incrementTimeout = setTimeout(() => {
+            incrementInterval = setInterval(incrementValue, 100);
+        }, 500);
+    };
+
+    const stopIncrementing = () => {
+        clearTimeout(incrementTimeout);
+        clearInterval(incrementInterval);
+    };
+
+    const startDecrementing = () => {
+        decrementTimeout = setTimeout(() => {
+            decrementInterval = setInterval(decrementValue, 100);
+        }, 500);
+    };
+
+    const stopDecrementing = () => {
+        clearTimeout(decrementTimeout);
+        clearInterval(decrementInterval);
+    };
+</script>
+
+<div class={containerClassString}>
+    {#if label}
+        <label class="label" for={$$restProps.id} style="color: {labelColor}"
+            >{$$restProps.id}</label
+        >
+    {/if}
+
+    <input
+        {color}
+        {value}
+        {disabled}
+        {step}
+        {min}
+        {max}
+        readonly={readOnly}
+        {...$$restProps}
+        id={label ? $$restProps.id : undefined}
+        type="number"
+        style={inputStyle}
+        on:input={handleInput}
+        on:input
+        on:focus
+        on:blur
+        on:change
+        on:click
+        on:keydown
+        on:keyup
+    />
+    <div class="controls">
+        <button
+            class="control up"
+            on:click={incrementValue}
+            on:mousedown={startIncrementing}
+            on:mouseup={stopIncrementing}
+            on:mouseleave={stopIncrementing}
+        >
+            <svg
+                xmlns="http://www.w3.org/2000/svg"
+                width="24"
+                height="24"
+                viewBox="0 0 24 24"
+                fill="none"
+                stroke={color}
+                stroke-width="2"
+                stroke-linecap="round"
+                stroke-linejoin="round"
+                class="feather feather-chevron-up"
+            >
+                <polyline points="18 15 12 9 6 15" />
+            </svg>
+        </button>
+        <button
+            class="control down"
+            on:click={decrementValue}
+            on:mousedown={startDecrementing}
+            on:mouseup={stopDecrementing}
+            on:mouseleave={stopDecrementing}
+        >
+            <svg
+                xmlns="http://www.w3.org/2000/svg"
+                width="24"
+                height="24"
+                viewBox="0 0 24 24"
+                fill="none"
+                stroke={color}
+                stroke-width="2"
+                stroke-linecap="round"
+                stroke-linejoin="round"
+                class="feather feather-chevron-down"
+            >
+                <polyline points="6 9 12 15 18 9" />
+            </svg>
+        </button>
+    </div>
+</div>
+
+<style lang="scss">
+    .input {
+        position: relative;
+        display: inline-flex;
+        flex-direction: column;
+        padding: 0;
+        &:focus-within {
+            .label-in {
```

Comment:
this label-in class I've seen a few places, I am not quite sure what 'label-in' conveys semantically if you could explain :)

Reply:
This is an oversight as I'm not including the labelIn prop for this component. See regular input under the label section of the UI to see a demonstration of how this feature works.